cscs-checks/libraries/math/dgemm.py

```diff
@@ -33,9 +33,22 @@ def __init__(self):
         }
 
         self.maintainers = ['AJ', 'VH', 'VK']
-        self.tags = {'diagnostic'}
+        self.tags = {'benchmark', 'diagnostic'}
 
     def setup(self, partition, environ, **job_opts):
+
+        if environ.name.startswith('PrgEnv-gnu'):
+            self.build_system.cflags += ['-fopenmp']
+        elif environ.name.startswith('PrgEnv-intel'):
+            self.build_system.cflags += [
+                '-qopenmp', '-DMKL_ILP64', '-I${MKLROOT}/include',
+                '-Wl,--start-group',
+                '${MKLROOT}/lib/intel64/libmkl_intel_ilp64.a',
+                '${MKLROOT}/lib/intel64/libmkl_intel_thread.a',
+                '${MKLROOT}/lib/intel64/libmkl_core.a',
+                '-Wl,--end-group',
+                '-liomp5', '-lpthread', '-lm', '-ldl']
```

Comment:
To be typically correct the options should be as follows:

```python
self.build_system.cppflags = ['-DMKL_ILP64', '-I{MKLROOT}/include']
self.build_system.cflags = ['-qopenmp']
self.build_system.ldflags = ['--the-rest']
```

Reply:
Thanks @vkarak, I have modified the code as you suggested